modules/Individual Needs/moduleFunctions.php

```diff
@@ -15,127 +15,117 @@
 
 You should have received a copy of the GNU General Public License
 along with this program. If not, see <http://www.gnu.org/licenses/>.
-*/
+ */
+
+use Gibbon\Domain\IndividualNeeds\INGateway;
+use Gibbon\Services\Format;
+use Gibbon\Tables\DataTable;
+use Gibbon\Domain\DataSet;
 
 //$mode can be blank or "disabled". $archive is a serialized array of values previously archived
 function printINStatusTable($connection2, $guid, $gibbonPersonID, $mode = '', $archive = '')
 {
-    $output = false;
-
-    try {
-        $dataDescriptors = array();
-        $sqlDescriptors = 'SELECT * FROM gibbonINDescriptor ORDER BY sequenceNumber, nameShort';
-        $resultDescriptors = $connection2->prepare($sqlDescriptors);
-        $resultDescriptors->execute($dataDescriptors);
-    } catch (PDOException $e) {
-        $output .= "<div class='error'>".$e->getMessage().'</div>';
-    }
-
-    try {
-        $dataSeverity = array();
-        $sqlSeverity = 'SELECT * FROM gibbonAlertLevel ORDER BY sequenceNumber, nameShort';
-        $resultSeverity = $connection2->prepare($sqlSeverity);
-        $resultSeverity->execute($dataSeverity);
-    } catch (PDOException $e) {
-        $output .= "<div class='error'>".$e->getMessage().'</div>';
-    }
-
-    if ($resultDescriptors->rowCount() < 1 or $resultSeverity->rowCount() < 1) {
-        $output .= "<div class='error'>";
-        $output .= __('Individual needs descriptors or severity levels have not been set.');
-        $output .= '</div>';
-    } else {
-        $descriptors = array();
-        $count = 0;
-        while ($rowDescriptors = $resultDescriptors->fetch()) {
-            $descriptors[$count][0] = $rowDescriptors['gibbonINDescriptorID'];
-            $descriptors[$count][1] = $rowDescriptors['name'];
-            $descriptors[$count][2] = $rowDescriptors['nameShort'];
-            $descriptors[$count][3] = $rowDescriptors['description'];
-            ++$count;
+    $output = "";
+    global $container;
```

Comment:
Hmmmm. I know this is needed, to make this particular function work, but I'm hesitant to start adding too many `global` calls to the codebase (although I have had to use them in the odd place too, as a transition to more oo code). Ideally, the whole function could be replaced by an object that is instantiated from the container.